### ISSUE TITLE
ECS Task Definition needs additional parameters for EFS Access Points

### ISSUE THREAD

**Issue by aberezin** [terraform-providers/terraform-provider-aws#13286]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

  AWS EFS access points are the most recent addition to EFS and are documented (here)[https://docs.aws.amazon.com/efs/latest/ug/efs-access-points.html].

  Currently, the aws_ecs_task_definition does not have an authorization_config for the efs_volume_configuration.  This means that an ecs_task_definition cannot create an EC2 nor Fargate instance connected to an EFS access point.   This ticket requests adding authorization_config.  

  It is unclear if terraform defined ECS Task Definition and efs_colume_configuration work with Fargate. See the references below.   It may be a separate issue but I need to use ECS Task Definition with Fargate.  This ticket is a necessary pre-condition.  To the extent that additional work is needed to make this all work with Fargate, that could be another ticket.

### New or Affected Resource(s)

* aws_ecs_task_definition

### Potential Terraform Configuration

```hcl
#requires https://github.com/terraform-providers/terraform-provider-aws/pull/11965
resource "aws_efs_access_point" "this" {
  file_system_id = "${aws_efs_file_system.foo.id}"
  subnet_id      = "${aws_subnet.alpha.id}"
}

resource "aws_ecs_task_definition" "service" {
  family                = "service"
  container_definitions = "${file("task-definitions/service.json")}"

  volume {
    name = "service-storage"

    efs_volume_configuration {
      file_system_id = aws_efs_file_system.fs.id
      root_directory = "/opt/data"
      transit_encryption = true #required for Access Points

      #this section is optional.  If included, the volume uses an access point
      authorization_config {
          accessPointId = aws_efs_access_point.this.id
          #Options are "ENABLED" or "DISABLED".  Default to "DISABLED"
          iam = "DISABLED" 
      }
    }
  }
}
```

### References

Requires [this PR](https://github.com/terraform-providers/terraform-provider-aws/pull/11965) for aws_efs_access_point resource.  

Requires [This PR](https://github.com/terraform-providers/terraform-provider-aws/pull/13136) for ECS transit encryption to EFS because to use an access point with ECS (Fargate or EC2), transit encryption is required.

[This ticket](https://github.com/terraform-providers/terraform-provider-aws/issues/12809) discusses the transit encryption request and makes some reference to needing authorizationConfig (which is this ticket).  

The released TF docs [here](https://www.terraform.io/docs/providers/aws/r/ecs_task_definition.html) say the following
> efs_volume_configuration - (Optional) Used to configure a EFS volume can be used only with an EC2 type task.

It may be a separate issue but I need to use ECS Task Definition with Fargate.

AWS task definition parameters for volumes are documented [here](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#volumes)

* #0000

**Comment by rickythefox** [comment]
Is there a workaround for this?

**Comment by aberezin** [comment]
There was no workaround and my solution was to abandon EFS access points in favor of the older style of EFS direct file mounts.  EFS Access Points are poorly supported across the board in TF and this is just one of the problems.  

On a somewhat unrelated note, I may abandon Fargate perhaps ECS altogether so the urgency of this ticket may be zero for me now.  Even if I stick with Fargate, it is not urgent to use Access Points because direct mounting of the EFS file system works well.

**Comment by durist** [comment]
I thought EFS access points are required to make Fargate work with EFS? From the announcement:

>

**Comment by alexgoldstone** [comment]
@durist Just to confirm this isn't just you, I have also not been able to access EFS from Fargate and believe it is because EFS access points are required.

**Comment by Jukie** [comment]
Fyi I've updated [my PR](#13136) to include the access point changes.

**Comment by durist** [comment]
It looks like access points were added in 2.63.0, but authorization_config still hasn't been added so they still don't work.

**Comment by bflad** [comment]
Support for this functionality has been merged and will release with version 2.68.0 of the Terraform AWS Provider, later this week. Thanks to @Jukie and others for the implementation. 👍
